src/js/view/controls/rightclick.js

```diff
@@ -98,7 +99,7 @@ export default class RightClick {
     }
 
     lazySetup() {
-        const html = rightclickTemplate(this.buildArray(), this.model.get('localization'));
+        const html = rightclickTemplate(this.buildArray(), this.model.get('localization'), this.shareOnRightClick);
```

Comment:
Use `buildArray()` to add items. `<li class="jw-reset jw-rightclick-item">` is repetitive and now that there are three types of items with different content, just add a type to each item in the menu array and handle the three types of items in the template `rightClickItem` method.

For example:
```js
// Handle all items with this method:
const menuItems = items.map(item => rightClickItem(item, localization))


const rightClickItem = (item, localization) => {
    const { link = '', title = '', featured, showLogo, type } = item;
    const item.logo = showLogo ? `<span class="jw-rightclick-logo jw-reset"></span>` : '';
    return `<li class="jw-reset jw-rightclick-item ${featured ? 'jw-featured' : ''}">${itemContentTypes[type](item, localization)}</li>`;
};

const itemContentTypes = {
    link({ link = '', title = '', logo }) => `<a href="${link}" class="jw-rightclick-link jw-reset" target="_blank">${logo}${title}</a>`,
    info(item, localization) => `<button type="button" class="jw-reset jw-rightclick-link jw-info-overlay-item">${localization.videoInfo}</button>`;
    share(item, localization) => /* you get the gist*/
};


```